tools/distpartitioning/data_shuffle.py

```diff
@@ -12,82 +12,109 @@
+                            ...,
+                            ["<path>/ntype0-name-<p-1>.txt", id_start<p-1>, id_end<p-1>]
+                        ]
+                },
+
+                ....,
+                "ntype<m-1>-name" : {
+                    "format" : "csv",
+                    "data" : [
+                        ["<path>/user-sup-ntype<m-1>-name-0.txt", 0, id_end0],
+                        ["<path>/user-sup-ntype<m-1>-name-1.txt", id_start1, id_end1],
+                        ...
+                        ["<path>/user-sup-ntype<m-1>-name-<p-1>.txt", id_start<p-1>, id_end<p-1>]
+                    ]
+                }
+            }
+
 
     Returns:
     --------
     dictionary : 
-        the input argument, node_data, is updated with the node data received by other processes
-        in the world.
-    """
-
-    input_list = []
-    send_sizes = []
-    recv_sizes = []
-    start = timer()
-    for i in np.arange(world_size):
-        send_idx = (node_data[constants.OWNER_PROCESS] == i)
-        idx = send_idx.reshape(node_data[constants.GLOBAL_NID].shape[0])
-        filt_data = np.column_stack((node_data[constants.NTYPE_ID][idx == 1], \
-                                node_data[constants.GLOBAL_TYPE_NID][idx == 1], \
-                                node_data[constants.GLOBAL_NID][idx == 1]))
-        if(filt_data.shape[0] <= 0): 
-            input_list.append(torch.empty((0,), dtype=torch.int64))
-            send_sizes.append(torch.empty((0,), dtype=torch.int64))
+        dictionary where keys are column names and values are numpy arrays, these arrays are generated by 
+        using information present in the metadata json file
+
+    '''
+    local_node_data = {}
+    gnid_start = 0
+    gnid_end = 0
+    for ntypeid in range(len(ntid_ntype_map)):
+        ntype_name = ntid_ntype_map[str(ntypeid)]
+        ntype_info = nid_schema_map[ntype_name]
+        type_start = int(ntype_info["data"][0][1])
+        type_end = int(ntype_info["data"][-1][2])
+
+        gnid_end += type_end
+
+        node_partid_slice = node_part_ids[gnid_start:gnid_end]
+        cond = node_partid_slice == rank
+        own_gnids = np.arange(gnid_start, gnid_end, dtype=np.int64)
+        own_gnids = own_gnids[cond]
+
+        own_tnids = np.arange(type_start, type_end, dtype=np.int64)
+        own_tnids = own_tnids[cond]
+        if constants.NTYPE_ID in local_node_data:
+            local_node_data[constants.NTYPE_ID].append(np.ones(own_gnids.shape, dtype=np.int64)*ntypeid)
         else:
-            input_list.append(torch.from_numpy(filt_data))
-            send_sizes.append(torch.tensor(filt_data.shape, dtype=torch.int64))
-        recv_sizes.append(torch.zeros((2,), dtype=torch.int64))
-    end = timer()
-    print('[Rank: ', rank, '] Preparing node_data to send out: ', timedelta(seconds=end - start))
+            local_node_data[constants.NTYPE_ID] = [np.ones(own_gnids.shape, dtype=np.int64)*ntypeid]
```

Comment:
to make code even simpler, you can create lists for each constant when `local_node_data` is created.
```
local_node_data = {
    constants.NTYPE_ID: [],
    ...
}
```

Reply:
Agreed. Now the code looks much cleaner and simpler to read. Thank you !!!